src/abstractarray.jl

```diff
@@ -227,7 +227,15 @@ for rot in [:rotl90, :rotr90]
     end
 end
 
-# TODO permutedims?
+# TODO permutedims? So far just the cases without perm:
+
+@generated function Base.permutedims(A::SMatrix{M,N}) where {M,N}
+    exs = permutedims([:(getindex(A,$i,$j)) for i in 1:M, j in 1:N])
+    return :(SMatrix{N,M}($(exs...)))
+end
+
+Base.permutedims(A::SVector{N}) where {N} = SMatrix{1,N}(A.data...)
+Base.permutedims(A::MVector{N}) where {N} = MMatrix{1,N}(A.data...)
```

Comment:
Calling `permutedims` on a normal `Array` gives an `Array` pointing at the same memory, and this does not. I'm not sure though if this data sharing is a part of the interface of `permutedims` or not? Docs are a bit unclear, in particular docstring for `permutedims(v::AbstractVector)` refers to the performed operation as "reshaping", and reshaping indicates data sharing.

Reply:
Right, I'm not sure what is best here. `permutedims(::Diagonal)` also shares memory, which makes me think that it should return the same type, but whether it's a view or a copy depends on what's convenient. 

`adjoint(::MMatrix)` also makes a copy; I guess `adjoint(::StaticVector)` has other reasons besides this copy/view issue to return a wrapper:
```
julia> using StaticArrays, LinearAlgebra

julia> d = Diagonal([1,2,3]);

julia> permutedims(d)
3×3 Diagonal{Int64, Vector{Int64}}:
 1  ⋅  ⋅
 ⋅  2  ⋅
 ⋅  ⋅  3

julia> ans[3,3] = 33;

julia> d
3×3 Diagonal{Int64, Vector{Int64}}:
 1  ⋅   ⋅
 ⋅  2   ⋅
 ⋅  ⋅  33

julia> m = @MMatrix [1 2; 3 4];

julia> m'
2×2 MMatrix{2, 2, Int64, 4} with indices SOneTo(2)×SOneTo(2):
 1  3
 2  4

julia> ans[2,2] = 22;

julia> m
2×2 MMatrix{2, 2, Int64, 4} with indices SOneTo(2)×SOneTo(2):
 1  2
 3  4

julia> SA[1,2,3]'
1×3 adjoint(::SVector{3, Int64}) with eltype Int64 with indices SOneTo(1)×SOneTo(3):
 1  2  3

julia> ans * SA[4,5,6]  # isa Number
32
```

Reply:
OK, so in the case of `MArray` memory sharing would definitely be quite inconvenient, and Base seems to be fine with not sharing memory: https://github.com/JuliaLang/julia/blob/27aaadb9b28e8457d86239e443f8e65d90e878c7/base/permuteddimsarray.jl#L124 . So, this approach to `MVector` and, similarly, `MMatrix` looks OK.

`SizedVector` and `SizedMatrix` should just call appropriate `permutedims` on the wrapped array, and wrap the result in a new `SizedArray`.

Reply:
For all higher dims, I thought `permutedims(::Array, ...)` never shared, until this `Diagonal` example. So the only concern was whether sharing was part of the contract for `::AbstractVector`, or just an accident of `::Vector`, and I think the conclusion is the latter.

Reply:
Yes, I think that's right.